modules/mono/glue/GodotSharp/GodotSharp/Core/Array.cs

```diff
@@ -342,24 +459,55 @@ public override string ToString()
     /// such as arrays or <see cref="List{T}"/>.
     /// </summary>
     /// <typeparam name="T">The type of the array.</typeparam>
-    public class Array<T> : IList<T>, ICollection<T>, IEnumerable<T>
+    [SuppressMessage("ReSharper", "RedundantExtendsListEntry")]
+    [SuppressMessage("Naming", "CA1710", MessageId = "Identifiers should have correct suffix")]
+    public sealed class Array<T> :
```

Comment:
Yeah, that's the reason I didn't implement `IDisposable` here yet. It's not clear disposing one will dispose the other. The solution would be for the underlying array to be a new managed instance, or for `Array<T>` to store the native value just like `Array`. In any case, I prefer to leave it outside the scope of this PR.

Reply:
So, currently, if you do `new Array<SomeType>(untypedArray)` this works like a cast instead of creating a `new Array`? Are they linked, so changing one would change the other? That's already counter-intuitive, considering `new Array<SomeType>(dotnetArray)` makes a copy of all elements. Or did I understand it wrong?
If that was a conversion operator or some `AsTyped<T>()` method I'd understand, but linking on a constructor is a bit weird.
If this behavior stays, perhaps clarifying this better in the docs is appropriate.

Reply:
@geekley use contributor chat or other channels, comments on an old already merged issue are unlikely to be read